pkgs/development/tools/language-servers/crystalline/default.nix

```
@@ -0,0 +1,45 @@
+{ lib
+, crystal
+, fetchFromGitHub
+, llvmPackages
+, openssl
+, makeWrapper
+}:
+
+crystal.buildCrystalPackage rec {
+  pname = "crystalline";
+  version = "0.9.0";
+
+  src = fetchFromGitHub {
+    owner = "elbywan";
+    repo = "crystalline";
+    rev = "v${version}";
```

you should do and remove the rec:
```suggestion
    rev = "v${finalAttrs.version}";
```

for more information see: 
    History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
    Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
    Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

---

I wasn't able to figure out how to use `finalAttrs` with `crystal.buildCrystalPackage` so I used let / in.

---

Oh `finalAttrs` probably isn't implemented in `buildCrystalPackage` my mistake.  Using the let binding is fine, you can also go back to `rec` then it's fine for me either way. Sorry :see_no_evil:

---

Thanks for the clarification. Will keep let/in then. Looking forward to the merge! :wink: